content/learn/book/the-game-loop/time-and-timers.md

```diff
@@ -0,0 +1,277 @@
+and then multiplying by the elapsed time.
+
+```rust, hide_lines=1-4
+# use bevy::prelude::*;
+# 
+# #[derive(Component)]
+# struct Player;
+#
+fn move_player(mut player_transform: Single<&mut Transform, With<Player>>, time: Res<Time>){
+    // At 60 FPS, this will be the same as before
+    const PLAYER_SPEED: f32 = 60.;
+    
+    player_transform.translation.x += PLAYER_SPEED * time.delta_secs_f32();
+}
+```
+
+This technique is commonly known as ["delta time"] among game devs, because physicists use "delta" to mean "a change in a quantity".
+
+["delta time"]: https://en.wikipedia.org/wiki/Delta_timing
+
+## Pausing and time control
+
+With all of our time-dependent logic driven by the delta time,
+we can start playing tricks with the value of [`Time`] to implement features like pausing and slowing down the game.
+To do this, we need to change how we account for [`Virtual`] (in-game) time:
+
+```rust
+use bevy::prelude::*;
+
+fn toggle_pause(mut time: ResMut<Time<Virtual>>) {
+    if time.is_paused() {
+        time.unpause();
+    } else {
+        time.pause();
+    }
+}
+
+#[derive(Event)]
+struct SetGameSpeed(f32);
+
+fn set_game_speed(mut time: ResMut<Time<Virtual>>, events: EventReader::<SetGameSpeed>){
+  if let Some(new_speed) = events.iter().last(){
+      time.set_relative_speed(new_speed.0);
+  }
+}
+```
+
+If your systems uniformly rely on [`Time`], this will effect your entire game:
+halting, slowing or speeding up animations, movement, projectiles, physics and so on.
+Alternatively, [states] and [run conditions] can be used to skip systems while the game is paused.
+
+[states]: ../control-flow/states.md
+[run conditions]: ../control-flow/run-conditions.md
+
+## Fixing your timestep
+
+Compensating for fluctating frame times using delta time is a great start.
+But as the timeless [fix your timestep article] explains for projects that require a higher level of stability and reproducibility,
+it's better to simply always advance time by a fixed amount.
+This is particularly important for physics and
+
+To understand how to work with fixed time in Bevy, we need to first learn a little bit about how [`Time`] actually works under the hood.
+As the excellent docs on [`Time`] explain, there's actually *three* distinct types of time being measured:
+
+- real time: the actual wall time
+  - use this for things like UI animations that you don't want to be affected by pausing
+- virtual time: the "in-game time"
+  - if you're using fixed time, this is useful for graphical effects
+  - otherwise, this is used for all of your gameplay logic as well
+- fixed time
+  - all of your gameplay logic should go here if you're using a fixed timestep approach
+
+When thinking about fixed time, it's important to clearly distinguish **frames** and **ticks**.
+A "frame" is one pass of the [`Main`] schedule, and corresponds to a single rendered frames.
+A "tick" is one pass of the [`FixedMain`] schedule, and corresponds to one iteration of gameplay logic.
```

Comment:
In my reactive frameworks, I advance the tick counter more often than this. Quill's "run to convergence" algorithm uses a loop which may advance the tick more than once in a single system execution (because I'm looking to detect fine-grained changes).

Reply:
What I was trying to express is that the phrase "corresponds to one iteration" may not be accurate. In most games, this is indeed true, but nothing prevents a game from advancing the tick counter more than once in this phase.